fe/src/main/java/org/apache/doris/analysis/SetVar.java

```diff
@@ -111,8 +113,22 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
         if (!(literalExpr instanceof LiteralExpr)) {
             throw new AnalysisException("Set statement does't support computing expr:" + literalExpr.toSql());
         }
- 
+
         result = (LiteralExpr)literalExpr;
+
+        if (variable.equalsIgnoreCase(SessionVariable.SQL_MODE)) {
+            // For the case like "set sql_mode = PIPES_AS_CONCAT"
+            if (result instanceof StringLiteral) {
+                String sqlMode = result.getStringValue();
+                result = new StringLiteral(SqlModeHelper.encode(sqlMode).toString());
+            }
+            // For the case like "set sql_mode = 3"
+            else if (result instanceof IntLiteral) {
+                String sqlMode = SqlModeHelper.decode(result.getLongValue());
+                result = new IntLiteral(SqlModeHelper.encode(sqlMode).toString(), Type.BIGINT);
```

Comment:
I think this is not needed. If decode execute successfully, you can just use result.